src/shared/utilities/collectionUtils.ts

```diff
@@ -160,3 +160,40 @@ export async function take<T>(sequence: AsyncIterable<T>, count: number): Promis
 
     return result
 }
+
+export interface IteratorForAWSCallParams<TRequest, TResponse> {
+    awsCall: (request: TRequest) => Promise<TResponse>
+    nextTokenNames: {
+        request: keyof TRequest
+        response: keyof TResponse
+    }
+    request: TRequest
+}
+
+/**
+ * Generates an iterator representing a paginated AWS call from a request and an AWS SDK call
+ * Each next() call will make a new request with the previous request's nextToken.
+ * @param params Iterator params
+ */
+export async function* getPaginatedAWSCallIter<TRequest, TResponse>(
```

Comment:
why is it spelled `AWS`? We use `Aws` in most places.

Reply:
Seems like we're split? The tree nodes use `AWS`. I'll swap it, although we should standardize at some point.